test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -62,6 +85,484 @@ def test_two_qubit_synthesis_to_basis(self, basis_gates):
+            backend_props=backend.properties(),
+            pulse_optimize=True,
+            natural_direction=True,
+        )
+        pm = PassManager([triv_layout_pass, unisynth_pass])
+        with self.assertRaises(TranspilerError):
+            pm.run(qc)
+
+    def test_two_qubit_pulse_optimal_none_optimal(self):
+        """Verify pulse optimal decomposition when pulse_optimize==None."""
+        # this assumes iswawp pulse optimal decomposition doesn't exist
+        backend = FakeVigo()
+        conf = backend.configuration()
+        qr = QuantumRegister(2)
+        coupling_map = CouplingMap([[0, 1], [1, 2], [1, 3], [3, 4]])
+        triv_layout_pass = TrivialLayout(coupling_map)
+        qc = QuantumCircuit(qr)
+        qc.unitary(random_unitary(4, seed=12), [0, 1])
+        unisynth_pass = UnitarySynthesis(
+            basis_gates=conf.basis_gates,
+            coupling_map=coupling_map,
+            backend_props=backend.properties(),
+            pulse_optimize=None,
+            natural_direction=True,
+        )
+        pm = PassManager([triv_layout_pass, unisynth_pass])
+        qc_out = pm.run(qc)
+        if isinstance(qc_out, QuantumCircuit):
+            num_ops = qc_out.count_ops()  # pylint: disable=no-member
+        else:
+            num_ops = qc_out[0].count_ops()
+        self.assertIn("sx", num_ops)
+        self.assertLessEqual(num_ops["sx"], 12)
+
+    def test_two_qubit_pulse_optimal_none_no_raise(self):
+        """Verify pulse optimal decomposition when pulse_optimize==None doesn't
+        raise when pulse optimal decomposition unknown."""
+        # this assumes iswawp pulse optimal decomposition doesn't exist
+        backend = FakeVigo()
+        conf = backend.configuration()
+        conf.basis_gates = [gate if gate != "cx" else "iswap" for gate in conf.basis_gates]
+        qr = QuantumRegister(2)
+        coupling_map = CouplingMap([[0, 1], [1, 2], [1, 3], [3, 4]])
+        triv_layout_pass = TrivialLayout(coupling_map)
+        qc = QuantumCircuit(qr)
+        qc.unitary(random_unitary(4, seed=12), [0, 1])
+        unisynth_pass = UnitarySynthesis(
+            basis_gates=conf.basis_gates,
+            coupling_map=coupling_map,
+            backend_props=backend.properties(),
+            pulse_optimize=None,
+            natural_direction=True,
+        )
+        pm = PassManager([triv_layout_pass, unisynth_pass])
+        try:
+            qc_out = pm.run(qc)
+        except QiskitError:
+            self.fail("pulse_optimize=None raised exception unexpectedly")
+        if isinstance(qc_out, QuantumCircuit):
+            num_ops = qc_out.count_ops()  # pylint: disable=no-member
+        else:
+            num_ops = qc_out[0].count_ops()
+        self.assertIn("sx", num_ops)
+        self.assertLessEqual(num_ops["sx"], 14)
+
+    @unittest.skipUnless(HAS_CPLEX, "cplex is required to run this test")
+    @unittest.skipUnless(HAS_DOCPLEX, "docplex is required to run this test")
```

Comment:
```suggestion
```

Reply:
Switching to Sabre allowed increasing the test size without needing non-free version of cplex. Thanks.